rust/src/dataset/scanner.rs

```diff
@@ -138,10 +169,29 @@ impl ScannerStream {
                             continue;
                         }
                     };
+                let fragment_id = frag.id;
+                assert!(fragment_id < 2 ^ 32);
                 for batch_id in 0..reader.num_batches() {
-                    tx.send(reader.read_batch(batch_id as i32).await)
-                        .await
-                        .unwrap();
+                    let batch = reader.read_batch(batch_id as i32).await.map(|b| {
+                        if with_row_id {
+                            // Add the meta "_rowid" column
+                            let mut columns = b.columns().to_vec();
+                            columns.push(Arc::new(UInt64Array::from(
+                                (row_count..row_count + b.num_rows() as u64 + fragment_id << 32)
+                                    .collect::<Vec<u64>>(),
+                            )));
+                            row_count += b.num_rows() as u64;
+                            RecordBatch::try_new(
```

Comment:
would it make sense to extend RecordBatch with append_column? so the schema and data change happens in one place.

Reply:
Sure thing. lemme file a new separate / small PR just for this change.